### ISSUE TITLE
Add support for login via GitLab, GitTea, Codeberg, Bitbucket, to add colab support and avoid GitHub walled garden.

### ISSUE THREAD

**Issue by iaxat** [zed-industries/zed#13296]
### Check for existing issues

- [X] Completed

### Describe the feature

Right now there is only support for GitHub, please add support for GitLab and GitTea.

**Comment by maxdeviant** [comment]
Could you please say more?

What features are you referring to that only support GitHub?

**Comment by samstride** [comment]
@maxdeviant , I think he means sign in with [Gitlab](https://docs.gitlab.com/ee/integration/oauth_provider.html) and [GitTea](https://docs.gitea.com/development/oauth2-provider) OAuth2 providers.

One thing to note is that Gitlab allows self hosting so will be good to configure your own auth provider instead of locking it down to the cloud instance of gitlab.

I am assuming that collaboration features will continue to work if sign in with an auth provider other than github is supported?

**Comment by nikepol** [comment]
I think it is about to open links to a commit, currently it is possible to see only a MR or commits from GitHub, but not a Gitlab or other providers

**Comment by iaxat** [comment]
The sign in for chat and collaboration are tied up to login by GitHub, and if GitTea, GitLab, Codeberg can be included as options that would be great.
I see GitLab is in pipeline.

**Comment by BillMeyerRSA** [comment]
Is it possible to add Bitbucket login support for collaboration as well?

**Comment by nathanchere** [comment]
Big +1 for this. I'm thankful you can use it without an account though. Github login = hard no.

**Comment by DerKarlos** [comment]
The Zed document:
Zed currently supports links to the hosted versions of [GitHub](https://github.com/), [GitLab](https://gitlab.com/), [Bitbucket](https://bitbucket.org/), [SourceHut](https://sr.ht/) and [Codeberg](https://codeberg.org/).

But how? The UI only uses GitHub. Any settings files?

**Comment by CHE1RON** [comment]
This would indeed be a great option, especially for all the folks with private GitLab instances 🤔

**Comment by sudo-nano** [comment]
I would love this as well. I'm trying to avoid depending on tech monopolies in my personal tech stack.

**Comment by iaxat** [comment]
Is there any update on this?

**Comment by soapyfrog** [comment]
I'm moving away from GitHub and find that the only way to sign in to Zed is via GitHub, so that's not going to work any more.
What's wrong with a straightforward username/password? or maybe a passkey?

Even Sign in with Apple/Google would be an improvement.  As it is, I cannot log in to Zed any more, so I'll be looking for development tools elsewhere.

**Comment by srinathv** [comment]
current progress?

**Comment by soapyfrog** [comment]
Zed is so focused on AI ,and AI fans probably don't even know alternatives to GitHub exist. So unless someone over age 40 puts up a PR for this, I don't see it ever getting done.

**Comment by tettoffensive** [comment]
This is really annoying. I would love to use the collaboration features of Zed, but my team doesn't use GitHub. We use Bitbucket.

**Comment by iaxat** [comment]
Yes, very annoying. We are working on projects using Codeberg since it offers free CI/CD for opensource projects. But we are stuck with GitHub. It does not look like Zed is interested at all.

**Comment by maxdeviant** [comment]
We are planning to add alternative sign-in methods (username/password, SSO) at some point in the future. No firm timeline on it right now, but it is on our radar.

**Comment by e-io** [comment]
@maxdeviant Thanks for the update! Looking forward to this feature. Many of us switched to Zed to get away from VS Code/Copilot/Microsoft, but found that paid AI features and Git integration still require an account which is under Microsoft's umbrella. 
It would be great if we could sign in with email just to pay for AI features or use alternative Git providers (GitLab, Codeberg). Thank you.

**Comment by iaxat** [comment]
Yeah do not want to get stuck with GitHub in any manner. Forgejo provides all the features, and I would very much like to not use anything from an evil org like MSFT.
